src/state/markets/fifo.rs

```diff
@@ -1400,11 +1412,28 @@ impl<
             / self.base_lots_per_base_unit;
         self.unclaimed_quote_lot_fees += inflight_order.quote_lot_fees;
 
-        Some(FIFORestingOrder::new(
-            current_trader_index as u64,
-            inflight_order.base_lot_budget,
-            inflight_order.last_valid_slot,
-            inflight_order.last_valid_unix_timestamp_in_seconds,
+        let best_price_on_opposite_book = self
```

Comment:
Why is this the best place to get `best_price_on_opposite_book`? It feels like it doesn't really belong here and maybe should just be grabbed at the point where we use it.

Reply:
The other natural place to do the check is when the order is actually inserted into the book. What I didn't like about that is that we already perform a cross check for PostOnly, so we would effectively be repeating the work.

Wouldn't be opposed to doing something like:
```
if matches!(order, OrderPacket::Limit(..) {
   // Check for cross
}
```
 
in latter half of place_order_inner